### PR TITLE
#98 fixed the sharing address book feature flag not being respected

### PR DESCRIPTION
**Pull request by rezk2ll**
https://user-images.githubusercontent.com/6764881/111460815-b4f94080-871c-11eb-8c82-cc2ae8b5a3db.mp4

### CODE REVIEW

File: src/esn.contact.libs/app/addressBook/addressbook-display.service.js [hunk 1]
```diff
@@ -46,11 +46,7 @@ require('./addressbook.constants.js');
         var addressbookDisplayShell = new match.displayShell(addressbookShell);
 
         if (options.includeActions) {
-          var actions = _.filter(match.actions, function(action) {
-            return action.name === 'Settings' ? options.isSharingAddressbookEnabled : true;
-          });
```
Comment (rezk2ll) [inline src/esn.contact.libs/app/addressBook/addressbook-display.service.js:51]: this action filter used to remove the settings menu button from the list when the address book sharing is disabled, which makes us lose (only accessible by URL, no button to go to the page ) the CalDAV URL field in the settings page ( and potentially the upcoming changes and added fields to the settings page ).

image


this filter had no test case that is why none was included in this PR, the existing cases cover what needs to be tested.

Comment (asmaNode) [reply src/esn.contact.libs/app/addressBook/addressbook-display.service.js:51]: I think related test must be updated here

Comment (rezk2ll) [reply src/esn.contact.libs/app/addressBook/addressbook-display.service.js:51]: the deleted changes had no tests before, the remaining code has test cases.

**Comment by huy-ta** [comment]
What are the weird vertical dots next to the "My contacts" address book?

**Comment by rezk2ll** [comment]
> What are the weird vertical dots next to the "My contacts" address book?

The address book action menu

**Review by huy-ta — APPROVED** [review]

**Review by renaudboyer — APPROVED** [review]

**Pull request merged**